test/libsolidity/syntaxTests/conversion/explicit_conversion_from_storage_array_ref.sol

```diff
@@ -0,0 +1,8 @@
+contract C {
+  int[10] x;
+  function f() public view {
+    int[](x);
+  }
+}
+// ----
+// TypeError: (59-67): Explicit type conversion not allowed from "int256[10] storage ref" to "int256[] storage ref".
```

Comment:
I'm actually surprised that it does not say `to "int256[] storage pointer`.

Reply:
**No, actually, it will always be a reference. The following is the evidence to show that the function copyForLocationIfReference, the type become a reference.**
```
1735                            resultType = ReferenceType::copyForLocationIfReference(dataLoc, resultType);
(gdb) p resultType->toString()
$1 = "int256[] memory"
(gdb) n
1736                            if (isNotAllowedArray || !argType->isExplicitlyConvertibleTo(*resultType))
(gdb) p resultType->toString()
$2 = "int256[] storage ref"
```

**Here is detailed Why**
```
(gdb) n
1735                            resultType = ReferenceType::copyForLocationIfReference(dataLoc, resultType);
(gdb) s
dev::solidity::ReferenceType::copyForLocationIfReference (_location=dev::solidity::DataLocation::Storage, _type=std::shared_ptr (count 2, weak 1) 0x1471df0) at /home/liang/projects/solidity/libsolidity/ast/Types.cpp:1509
1509    {
(gdb) n
1510            if (auto type = dynamic_cast<ReferenceType const*>(_type.get()))
(gdb) 
1511                    return type->copyForLocation(_location, false);
(gdb) s
dev::solidity::ArrayType::copyForLocation (this=0x1471df0, _location=dev::solidity::DataLocation::Storage, _isPointer=false) at /home/liang/projects/solidity/libsolidity/ast/Types.cpp:1846
1846    {
(gdb) n
1847            auto copy = make_shared<ArrayType>(_location);
(gdb) 
1848            copy->m_isPointer = _isPointer;
(gdb) p _isPointer
$3 = false
```

Reply:
@chriseth @ekpyron 
The copyForLocationIfReference will always pass false for _isPointer whenever it can dynamically convert to a reference type. **And, ArrayType is inherit from ReferenceType, so it will be alway true for array type.**
The evidense: **class ArrayType: public ReferenceType**